### PR TITLE
Implement ppc assembler plugin

### PR DESCRIPTION
**Pull request by eagleoflqj**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

`rasm2 -L` shows no assembler for ppc. Without an assembler, ragg2 cannot run on ppc.

**Test plan**

Install binutils-powerpc64le-linux-gnu (available on x64 and arm64) and `export R2_PPC_AS=powerpc64le-linux-gnu-as` (needn't do anything on ppc64). Then `r2r db/tools/rasm2`.

Edit for 260e0b5:
8 tests are added. 2 of them use ppc.as (have to be marked as broken). 2 have the argument and result swapped.
The other 4 use arm.as. No comparison this time, because the disassembler has some bugs. I will report them in another issue.
On ppc64: compared to [original build](https://travis-ci.com/github/eagleoflqj/radare2/jobs/440092506#L1053), [this build](https://travis-ci.com/github/eagleoflqj/radare2/jobs/441975186#L939) adds 2 OKs and 6 FXs.
On arm64: compared to [original build](https://travis-ci.com/github/eagleoflqj/radare2/jobs/440092508#L1071), [this build](https://travis-ci.com/github/eagleoflqj/radare2/jobs/441975188#L958) adds 2 OKs and 6 FXs.

**Closing issues**

...

### CODE REVIEW

**Review by wargio — APPROVED** [review]
LGTM

File: libr/asm/p/asm_ppc_as.c [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	char *as = "as";
+#else
+	char *as = "";
+#endif
+	char *user_as = r_sys_getenv (ASSEMBLER);
+	if (user_as) {
+		as = user_as;
+	}
+	if (R_STR_ISEMPTY (as)) {
+		eprintf("Please set "ASSEMBLER" env to define a ppc assembler program\n");
```
Comment (trufae) [inline libr/asm/p/asm_ppc_as.c:21]: Missing space

File: libr/asm/p/asm_ppc_as.c [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		close (ifd);
+		close (ofd);
+		return -1;
+	}
+	const size_t asm_buf_len = strlen (asm_buf);
+	const bool success = write (ifd, asm_buf, asm_buf_len) == asm_buf_len;
+	close (ifd);
+	free (asm_buf);
+	if (!success) {
+		free (user_as);
```
Comment (trufae) [inline libr/asm/p/asm_ppc_as.c:58]: Use goto. This is too soaguetti and easy to make mistakes

File: libr/asm/p/asm_ppc_as.c [hunk 3]
```diff
@@ -0,0 +1,132 @@
+/* radare - LGPL - Copyright 2015-2020 pancake */
```
Comment (trufae) [inline libr/asm/p/asm_ppc_as.c:1]: The copyright is wrong. This file wasnt born in 2015 and it was not commited by me. Put your name

**Review by trufae — CHANGES_REQUESTED** [review]
Also i think all those custom binutils should have an option to pull and run the corresponding docker image.

Also it will be good to reuse the code that is common between all the binutil assemblers via system instead of copypasting

File: test/db/tools/rasm2 [hunk 4]
```diff
@@ -600,3 +600,30 @@ EXPECT=<<EOF
 mr r10, r4
 EOF
 RUN
+
+NAME=rasm2 -a arm.as -b 16 "wfe"
```
Comment (trufae) [inline test/db/tools/rasm2:604]: Tests should require external stuff to be installed i dont expect the ci to have binutils for arm, ppc, mips..

Maybe those tests would require a condition or add those toolchains

Comment (eagleoflqj) [reply test/db/tools/rasm2:604]: They are (and have to be due to lack of binutils on win/mac/s390x ...) marked broken. I've added needed packages to arm64 and ppc on travis.

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
It looks good to me and also a good idea to extract the common code but please do a rebase.

**Review by ret2libc — APPROVED** [review]

**Pull request merged**